src/goto-instrument/cover.cpp

```diff
@@ -1445,13 +1448,32 @@ bool instrument_cover_goals(
       c=coverage_criteriont::COVER;
     else
     {
-      msg.error() << "unknown coverage criterion" << messaget::eom;
+      msg.error() << "unknown coverage criterion "
+                  << criterion_string
+                  << messaget::eom;
       return true;
     }
 
     criteria.insert(c);
   }
 
+  msg.status() << "Rewriting existing assertions as assumptions"
+               << messaget::eom;
+
+  if(!keep_assertions)
```

Comment:
Wouldn't it be good to issue a warning in case `keep_assertions && criteria.size()>1`?